src/Components/ThemesMenu/ThemesMenu.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import Button from "../Button/Button";
+import "./ThemesMenu.css";
+
+export function ThemesMenu({
+  currentTheme,
+  onChange,
+  themes,
+  onAddTheme,
+  onEditTheme,
+  onDeleteTheme,
+}) {
+  const [newThemeName, setNewThemeName] = useState("");
+  const [isEditingTheme, setIsEditingTheme] = useState(false);
+  const [editThemeName, setEditThemeName] = useState(currentTheme.name);
+  const [isDelitingTheme, setIsDeletingTheme] = useState(false);
+
+  const handleThemeNameChange = (event) => {
+    setNewThemeName(event.target.value);
+  };
+
+  const handleEditThemeNameChange = (event) => {
+    setEditThemeName(event.target.value);
+  };
+
+  const handleEditClick = () => {
+    setIsEditingTheme(true);
+    setEditThemeName(currentTheme.name);
+  };
```

Comment:
update the handlers based on using single state: ` const handleEditClick = () => {

    if (currentTheme.id !== themes[0].id) {

      setThemeState({

        ...themeState,

        isEditingTheme: true,

        editThemeName: currentTheme.name,

      });

    }

  };`